demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/NotificationReceiver.java

```diff
@@ -3,12 +3,34 @@
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.widget.Toast;
+
+import com.novoda.downloadmanager.lib.DownloadManager;
 
 public class NotificationReceiver extends BroadcastReceiver {
 
     @Override
     public void onReceive(Context context, Intent intent) {
+        int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
+        Toast.makeText(context, "Clicked on a " + getStatusMessage(statuses) + " notification", Toast.LENGTH_LONG).show();
         context.startActivity(new Intent(context, MainActivity.class).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK));
     }
 
+    private String getStatusMessage(int[] statuses) {
+        switch (statuses[0]) {
```

Comment:
What about a `StatusMessage` enum with a `from(int status)` method? the `toString()` is given by free.